packages/opentelemetry-types/src/trace/instrumentation/Plugin.ts

```diff
@@ -32,9 +32,60 @@ export interface Plugin<T = any> {
     moduleExports: T,
     tracer: Tracer,
     logger: Logger,
-    config?: { [key: string]: unknown }
+    config?: Partial<PluginConfig>
   ): T;
 
   /** Method to disable the instrumentation  */
   disable(): void;
 }
+
+export interface PluginConfig {
+  /**
+   * Whether to enable the plugin.
+   * @default true
+   */
+  enabled: boolean;
+
+  /**
+   * Path of the trace plugin to load.
+   * @default '@opentelemetry/plugin-http' in case of http.
+   */
+  path: string;
+
+  /**
+   * Request methods that match any string in ignoreMethods will not be traced.
+   */
+  ignoreMethods: string[];
+
+  /**
+   * URLs that partially match any regex in ignoreUrls will not be traced.
+   * In addition, URLs that are _exact matches_ of strings in ignoreUrls will
+   * also not be traced.
+   */
+  ignoreUrls: Array<string | RegExp>;
```

Comment:
Should this be part of the general `PluginOptions` or should it be specific to the HTTP related plugins?

Reply:
Hmm, good question. WDYT?

Reply:
In my opinion it should be in the plugin config for the HTTP(S) plugins. In general I think there should be a mechanism for plugins to set their own option types. Is that supported now?

Reply:
No, currently this is not supported for now but will revisit again in the future.

Reply:
OK, then given that I think keeping this in the common plugin config makes sense. Should it be marked optional though `ignoreUrls?: Array<string | RegExp>`?